sphinx/ext/intersphinx/_load.py

```diff
@@ -21,55 +21,100 @@
 
     from sphinx.application import Sphinx
     from sphinx.config import Config
-    from sphinx.ext.intersphinx._shared import InventoryCacheEntry
+    from sphinx.ext.intersphinx._shared import (
+        IntersphinxMapping,
+        InventoryCacheEntry,
+        InventoryLocation,
+        InventoryName,
+        InventoryURI,
+    )
     from sphinx.util.typing import Inventory
 
 
 def normalize_intersphinx_mapping(app: Sphinx, config: Config) -> None:
-    for key, value in config.intersphinx_mapping.copy().items():
+    # URIs should NOT be duplicated, otherwise different builds may use
+    # different project names (and thus, the build are no more reproducible)
+    # depending on which one is inserted last in the cache.
+    seen: dict[InventoryURI, InventoryName] = {}
+
+    for name, value in config.intersphinx_mapping.copy().items():
+        if not isinstance(name, str):
+            LOGGER.warning(__('intersphinx identifier %r is not string. Ignored'),
+                           name, type='intersphinx', subtype='config')
+            del config.intersphinx_mapping[name]
+            continue
+
+        # ensure that intersphinx projects are always named
+        if not name:
+            LOGGER.warning(
+                __('ignoring empty intersphinx identifier'),
+                type='intersphinx', subtype='config',
+            )
+            del config.intersphinx_mapping[name]
+            continue
+
+        if not isinstance(value, (tuple, list)):
+            LOGGER.warning(
+                __('intersphinx_mapping[%r]: expecting a tuple or a list, got: %r; ignoring.'),
+                name, value, type='intersphinx', subtype='config',
+            )
+            del config.intersphinx_mapping[name]
+            continue
+
         try:
-            if isinstance(value, (list, tuple)):
-                # new format
-                name, (uri, inv) = key, value
-                if not isinstance(name, str):
-                    LOGGER.warning(__('intersphinx identifier %r is not string. Ignored'),
-                                   name)
-                    config.intersphinx_mapping.pop(key)
-                    continue
+            uri, inv = value
+        except Exception as exc:
+            LOGGER.warning(
+                __('Failed to read intersphinx_mapping[%s], ignored: %r'),
+                name, exc, type='intersphinx', subtype='config',
+            )
+            del config.intersphinx_mapping[name]
+            continue
```

Comment:
I think we should either (a) raise a ConfigError immediately, or (b) collect all errors and then raise a ConfigError with all the errors.

Reply:
Mmh. This works for me. I'm going to eat and will switch to gaming mode so I won't work on this anymore (but I'll work on it tomorrow). 

- raising the error immediately: less code and check in tests, but incremental debugging may be needed (multiple rebuilds if you have multiple errors).
- deferring the exception: current flow can be kept (just adding some `.append` and a final join in the final message), and you see all errors in one go.

I think deferring the exception is fine.

Reply:
I've pushed an initial implementation of this -- an alternative would be to instead log the errors at the test/validate site and increment an error counter, which would allow for more specific error messages. I also haven't tested with ``-W`` yet. I'm away most of today but should be able to look again this evening.

A